test/e2e/README.md

```diff
@@ -1,128 +1,83 @@
-export REGISTRY='...'
-
-# The version of the NMI image to test. Defaults to `1.4`.
-export NMI_VERSION='...'
-
-# The version of the MIC image to test. Defaults to `1.3`.
-export MIC_VERSION='...'
+# login as a user
+az login
 
-# The version of the identity validator to test. Defaults to `1.4`.
-export IDENTITY_VALIDATOR_VERSION='...'
+export SUBSCRIPTION_ID="<SubscriptionID>"
+export RESOURCE_GROUP="<ClusterResourceGroup>"
+export KEYVAULT_NAME="my-keyvault"
+export KEYVAULT_SECRET_NAME="test-secret"
 
-# The mode in which NMI is run. Allowed values: standard, managed. Default is standard
-export NMI_MODE='...'
+./test/e2e/setup-e2e.sh
 ```
 
-If you are using system asssigned identity cluster, please set the following variable:
+The E2E test suite extracts runtime configurations through environment variables. Below is a list of environment variables to set before running the E2E test suite.
+
+| Variable                       | Description                                                                            |
+|--------------------------------|----------------------------------------------------------------------------------------|
+| `SUBSCRIPTION_ID`              | The Azure subscription ID.                                                             |
+| `RESOURCE_GROUP`               | The resource group of your Azure Kubernetes cluster.                                   |
+| `AZURE_CLIENT_ID`              | The client ID of your service principal.                                               |
+| `AZURE_CLIENT_SECRET`          | The client secret of your service principal.                                           |
+| `AZURE_TENANT_ID`              | The Azure tenant ID.                                                                   |
+| `KEYVAULT_NAME`                | The Azure KeyVault name.                                                               |
+| `KEYVAULT_SECRET_NAME`         | The name of the secret stored in the Azure KeyVault.                                   |
+| `KEYVAULT_SECRET_VERSION`      | The version of the secret stored in the Azure KeyVault.                                |
+| `MIC_VERSION`                  | The MIC vesrion.                                                                       |
+| `NMI_VERSION`                  | The NMI version.                                                                       |
+| `IDENTITY_VALIDATOR_VERSION`   | The identity validator version                                                         |
+| `SYSTEM_MSI_CLUSTER`           | Set to `true` if you are using an Azure cluster with system-assigned identity enabled. |
+| `ENABLE_SCALE_FEATURES`        | Set to `true` if you want to enable the scale features.                                |
+| `IMMUTABLE_IDENTITY_CLIENT_ID` | The client ID of the immutable user-assigned identity.                                 |
```

Comment:
`The client ID of the immutable user-assigned identity created by running setup.sh`